build.gradle.kts

```diff
@@ -66,8 +61,19 @@ repositories {
 
 allprojects {
     repositories {
-        mavenLocal()
+        System.getenv("CODEARTIFACT_URL")?.let {
+            println("Using CodeArtifact proxy: $it")
+            maven {
+                url = uri(it)
+                credentials {
+                    username = "aws"
+                    password = System.getenv("CODEARTIFACT_AUTH_TOKEN")
+                }
+            }
+        }
+        gradlePluginPortal()
         mavenCentral()
+        mavenLocal()
```

Comment:
mevenLocal should go first, right? Or does the order not matter

Reply:
Why does it need to go first?

Reply:
Order matters for the order it is checked in, local going first only matters if you need to use a custom / patched version to override what would be in maven central.

Reply:
Yes, which means we should always put it first just in case we need it? Like if we want to test a common package before?

Reply:
Fair